src/App.php

```diff
@@ -311,12 +318,14 @@ function ( $wikiLang ) use ( $slim ) {
 					$slim->authManager->logout();
 					$slim->redirect( $slim->urlFor( 'root' ) );
 				} )->name( 'logout' );
-				$slim->get( ':wikiLang/loadmore', function ( $wikiLang ) use ( $slim ) {
-					$page = new Controllers\CopyPatrol( $slim );
-					$page->setDao( $this->getPlagiabotDao() );
-					$page->setWikiDao( $this->getWikiDao( $wikiLang ) );
-					$page();
-				} )->name( 'loadmore' )->setConditions( $routeConditions );
+				$slim->get( ':wikiLang/loadmore', $middleware['valid-wikiLang'],
+					function ( $wikiLang ) use ( $slim ) {
+						$page = new Controllers\CopyPatrol( $slim );
+						$page->setDao( $this->getPlagiabotDao() );
+						$page->setWikiDao( $this->getWikiDao( $wikiLang ) );
+						$page();
+					}
+				)->name( 'loadmore' )->setConditions( $routeConditions );
```

Comment:
Yeah, I actually refactored all the routes twice, before returning to the status quo; seemed simpler. But I'm absolutely more in favour of more-readable routes, so will have another crack at it if you think it's a good idea too. :-)